src/AdminConnection.ts

```diff
@@ -1,1213 +1,1218 @@
+							resolve(
+								getObjectViewResultToArray<ioBroker.UserObject>(
+									doc,
+								),
+							);
+						}
+					},
+				);
+			},
+		});
+	}
+
+	/**
+	 * Get the list of all groups.
+	 * @param update Force update.
+	 */
+	getGroups(update?: boolean): Promise<ioBroker.GroupObject[]> {
+		return this.request({
+			cacheKey: "groups",
+			forceUpdate: update,
+			// TODO: check if this should time out
+			commandTimeout: false,
+			requireAdmin: true,
+			executor: (resolve, reject) => {
+				this._socket.emit(
+					"getObjectView",
+					"system",
+					"group",
+					{
+						startkey: "system.group.",
+						endkey: "system.group.\u9999",
+					},
+					(err, doc) => {
+						if (err) {
+							reject(err);
+						} else {
+							resolve(
+								getObjectViewResultToArray<ioBroker.GroupObject>(
+									doc,
+								),
+							);
+						}
+					},
+				);
+			},
+		});
+	}
+
+	/**
+	 * Rename a group.
+	 * @param id The id.
+	 * @param newId The new id.
+	 * @param newName The new name.
+	 */
+	renameGroup(
+		id: string,
+		newId: string,
+		newName: ioBroker.StringOrTranslated,
+	): Promise<void> {
+		return this.request({
+			// TODO: check if this should time out
+			commandTimeout: false,
+			requireAdmin: true,
+			executor: async (resolve) => {
+				const groups = await this.getGroups(true);
+				// renaming a group happens by re-creating the object under a different ID
+				const subGroups = groups.filter((g) =>
+					g._id.startsWith(`${id}.`),
+				);
+				// First do this for all sub groups
+				for (const group of subGroups) {
+					const oldGroupId = group._id;
+					const newGroupId = newId + group._id.substring(id.length);
+					group._id = newGroupId;
+
+					// Create new object, then delete the old one if it worked
+					await this.setObject(newGroupId, group);
+					await this.delObject(oldGroupId);
+				}
+				// Then for the parent group
+				const parentGroup = groups.find((g) => g._id === id);
+				if (parentGroup) {
+					const oldGroupId = parentGroup._id;
+					parentGroup._id = newId;
+					if (newName !== undefined) {
+						(parentGroup.common as any) ??= {};
+						parentGroup.common.name = newName as any;
+					}
+
+					// Create new object, then delete the old one if it worked
+					await this.setObject(newId, parentGroup);
+					await this.delObject(oldGroupId);
+				}
+
+				resolve();
+			},
+		});
+	}
+
+	/**
+	 * Get the host information.
+	 * @param host
+	 * @param update Force update.
+	 * @param timeoutMs optional read timeout.
+	 */
+	getHostInfo(
+		host: string,
+		update?: boolean,
+		timeoutMs?: number,
+	): Promise<any> {
+		host = normalizeHostId(host);
+		return this.request({
+			cacheKey: `hostInfo_${host}`,
+			forceUpdate: update,
+			commandTimeout: timeoutMs,
+			requireAdmin: true,
+			executor: (resolve, reject, timeout) => {
+				this._socket.emit(
+					"sendToHost",
+					host,
+					"getHostInfo",
+					null,
+					(data) => {
+						if (timeout.elapsed) return;
```

Comment:
Here and in ~28 other cases we shouldn't care about the timeout. A promise can never complete twice and also clearing an already cleared or elapsed timeout doesn't matter. For the `clearTimeout()` see my other comment. So: simply remove this line completely.

Reply:
You're right that it doesn't matter here per se. But I guarantee that someone at some point changes the code below in a way that it will matter. I'd like to prevent doing unnecessary work by just keeping it that way for now.

Most of the other cases that don't have a timeout now should probably have one too. We could then refactor the executor method in a way that this check is handled outside the socket callback.